Bricks/mq/inmemory/mq.h

```diff
@@ -53,107 +53,26 @@ SOFTWARE.
 
-#include "consumer.h"
+#include "../interface/interface.h"
 
 namespace bricks {
 namespace mq {
 
-namespace mmq {
-
-template <typename T_CONSUMER, typename T_MESSAGE>
-constexpr bool HasSimpleOnMessage(char) {
-  return false;
-}
-
-template <typename T_CONSUMER, typename T_MESSAGE>
-constexpr auto HasSimpleOnMessage(int)
-    -> decltype(std::declval<T_CONSUMER>().OnMessage(std::move(std::declval<T_MESSAGE>())), bool()) {
-  return true;
-}
-
-template <typename T_CONSUMER, typename T_MESSAGE>
-constexpr bool HasExtendedOnMessage(char) {
-  return false;
-}
-
-template <typename T_CONSUMER, typename T_MESSAGE>
-constexpr auto HasExtendedOnMessage(int)
-    -> decltype(std::declval<T_CONSUMER>().OnMessage(std::move(std::declval<T_MESSAGE>()), 0u), bool()) {
-  return true;
-}
-
-template <typename T_CONSUMER, typename T_MESSAGE, bool HAS_SIMPLE_ON_MESSAGE>
-struct ExportMessageImpl {};
-
-template <typename T_CONSUMER, typename T_MESSAGE>
-struct ExportMessageImpl<T_CONSUMER, T_MESSAGE, true> {
-  static void OnMessage(T_CONSUMER& consumer, T_MESSAGE&& message, size_t) {
-    consumer.OnMessage(std::move(message));
-  }
-};
-
-template <typename T_CONSUMER, typename T_MESSAGE>
-struct ExportMessageImpl<T_CONSUMER, T_MESSAGE, false> {
-  static void OnMessage(T_CONSUMER& consumer, T_MESSAGE&& message, size_t dropped_count) {
-    consumer.OnMessage(std::move(message), dropped_count);
-  }
-};
-
-template <typename T_CONSUMER, typename T_MESSAGE>
-void ExportMessage(T_CONSUMER& consumer, T_MESSAGE&& message, size_t dropped_count) {
-  // Note that the allowed syntax for `OnMessage()` is:
-  //   either `OnMessage(const T_MESSAGE& message [, size_t dropped_count]);`
-  //   or     `OnMessage(T_MESSAGE&& message [, size_t dropped_count]);`
-  // If your class defines `OnMessage()` differently, for example, accepting a non-const non-rvalue reference
-  // as the first parameter, the `static_assert()` on the next line will fail.
-  static_assert(HasSimpleOnMessage<T_CONSUMER, T_MESSAGE>(0) || HasExtendedOnMessage<T_CONSUMER, T_MESSAGE>(0),
-                "There must be at least one implementation of `OnMessage()` defined in the consumer.");
-  static_assert(HasSimpleOnMessage<T_CONSUMER, T_MESSAGE>(0) != HasExtendedOnMessage<T_CONSUMER, T_MESSAGE>(0),
-                "There must be exactly one implementation of `OnMessage()` defined in the consumer.");
-  ExportMessageImpl<T_CONSUMER, T_MESSAGE, HasSimpleOnMessage<T_CONSUMER, T_MESSAGE>(0)>::OnMessage(
-      consumer, std::forward<T_MESSAGE>(message), dropped_count);
-}
-
-}  // namespace mmq
-
-template <typename MESSAGE,
-          typename CONSUMER = mmq::ProcessMessageConsumer<MESSAGE>,
-          size_t DEFAULT_BUFFER_SIZE = 1024,
-          bool DROP_ON_OVERFLOW = false>
+template <typename MESSAGE, typename CONSUMER, size_t DEFAULT_BUFFER_SIZE = 1024, bool DROP_ON_OVERFLOW = false>
 class MMQ final {
  public:
   // Type of entries to store, defaults to `std::string`.
```

Comment:
Just noticed that this comment is outdated, we don't have default now. Could you remove it pls?